### ISSUE TITLE
[question] Building editable dependencies

### ISSUE THREAD

**Issue by marausch** [conan-io/conan#15563]
### What is your question?

Hi!
As shown in the [documentation](https://docs.conan.io/2/tutorial/developing_packages/editable_packages.html#building-editable-dependencies), to do a full ordered build of a project including the editable dependencies we need to add the `--build=editable` argument to the `conan build` command (BTW, the [reference](https://docs.conan.io/2/reference/commands/build.html) doesn't mention it).
I'm interested about the reason why `--build=missing` wouldn't have been a good implementation choice to achieve the same (AFAIU, Conan should be able to detect if a dependency is in editable mode without the need to specify it on the command line).
Possibly a stupid question... I apologise. ;)
TIA

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @marausch 

No need to apologize, no question is ever stupid!

> I'm interested about the reason why --build=missing wouldn't have been a good implementation choice to achieve the same (AFAIU, Conan should be able to detect if a dependency is in editable mode without the need to specify it on the command line).
Possibly a stupid question... I apologise. ;)

The reasons is that the semantics and intended behavior are different. When you have a project with some dependencies in ``editable`` mode, it is very valid to want to build only those, but not accidentally starting a build from source from other packages that are in the Conan cache, something that would happen with ``--build=missing``.

Also when a package is in editable mode, it is perfectly possible to have a binary for it locally, so the binary will be existing and not missing. It is possible that the editable has changed the source code and not build yet, but that would be more equivalent to a new revision than to a missing binary. There are users that their ``build()`` will not fully work in ``editable`` mode and they will rely on manual build commands. They still need a mechanism to build the cache missing binaries without triggering a force build of the packages in ``editable`` mode.

In summary, adding a new ``--build=editable`` allows supporting all different flows that users have, while using the ``--build=missing`` to also build the editables will not have enough flexibility for some users.
